test/python/WMCore_t/JobStateMachine_t/ChangeState_t.py

```diff
@@ -644,6 +647,7 @@ def testFWJRInputFileTruncation(self):
         locationAction = self.daoFactory(classname = "Locations.New")
         locationAction.execute("site1", pnn = "T2_CH_CERN")
 
+        self.specGenerator.createRandomProductionSpec("spec.xml", "file")
```

Comment:
cannot use createRandomProductionSpec, I would remove this line and put the code in init.